### PR TITLE
Add SwiftToolchain class

### PR DESCRIPTION
**Pull request by adam-fowler**
Stores details about the current swift toolchain. Collects various functions into one class

### CODE REVIEW

File: src/toolchain/toolchain.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    private static async getDeveloperDir(): Promise<string | undefined> {
+        try {
+            switch (process.platform) {
+                case "darwin": {
+                    const { stdout } = await execFile("xcode-select", ["-p"]);
+                    return stdout.trimEnd();
+                }
+
+                case "win32": {
+                    const developerPath = process.env.DEVELOPER_DIR;
```
Comment (stevapple) [inline src/toolchain/toolchain.ts:80]: This should be exactly what `DeveloperDir` is on Windows, so I think the naming is improper. Change it back into `getXCTestDir` instead?

**Review by stevapple — CHANGES_REQUESTED** [review]

File: src/toolchain/toolchain.ts (hunk 1 above)
Comment (adam-fowler) [reply src/toolchain/toolchain.ts:80]: Sorry yeah I got that one wrong.

Comment (adam-fowler) [reply src/toolchain/toolchain.ts]: I've split it into two functions `getDeveloperDir` and `getXCTestPath`

Comment (adam-fowler) [reply src/toolchain/toolchain.ts]: @stevapple would you be able to verify I haven't broken Windows

**Comment by stevapple** [comment]
It works:)

**Review by stevapple — APPROVED** [review]

File: src/toolchain/toolchain.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VSCode Swift open source project
+//
+// Copyright (c) 2021 the VSCode Swift project authors
```
Comment (0xTim) [inline src/toolchain/toolchain.ts:5]: We should update the script to update the date to be 2021-2022

**Pull request merged**